zkevm-circuits/src/root_circuit/pcd_aggregation.rs

```diff
@@ -0,0 +1,375 @@
+//! The Root circuit implementation.
+use halo2_proofs::{
+    circuit::{Layouter, SimpleFloorPlanner},
+    halo2curves::bn256::{Bn256, Fr},
+    plonk::{Circuit, ConstraintSystem, Error},
+    poly::kzg::commitment::ParamsKZG,
+};
+use itertools::Itertools;
+use maingate::{MainGateInstructions, RangeInstructions};
+use snark_verifier_sdk::{CircuitExt, LIMBS};
+use std::fmt;
+
+pub use snark_verifier::system::halo2::{compile, Config};
+use snark_verifier_sdk::{
+    halo2::aggregation::{AccumulationSchemeSDK, AggregationCircuit, AggregationConfig},
+    Snark,
+};
+
+// TODO: move this to snark-verifier-sdk
+/// AggregationType is used to specify which accumulation scheme to use.
+#[derive(Clone, Copy, Debug)]
+pub enum AccumulationSchemeType {
+    /// using GWC
+    GwcType,
+    /// using SHPLONK
+    ShplonkType,
+}
+
+impl From<AccumulationSchemeType> for u64 {
+    fn from(t: AccumulationSchemeType) -> u64 {
+        match t {
+            AccumulationSchemeType::GwcType => 0,
+            AccumulationSchemeType::ShplonkType => 1,
+        }
+    }
+}
+
+impl From<u64> for AccumulationSchemeType {
+    fn from(t: u64) -> AccumulationSchemeType {
+        match t {
+            0 => AccumulationSchemeType::GwcType,
+            1 => AccumulationSchemeType::ShplonkType,
+            _ => panic!("Invalid aggregation type"),
+        }
+    }
+}
+
+/// PCDAggregationCircuit for aggregating various sub circuits into a smaller proof.
+#[derive(Clone)]
+pub struct PCDAggregationCircuit<AS>
```

Comment:
PCD = Proof Carrying Data?

Reply:
Yes, name suggestion? since rootXXX, aggregationXXX already everywhere :)

Reply:
I was just wondering. :)

I guess it's the most standard way to expose all the public inputs so I don't know what a good name could be. We may as well call it the `TaikoAggregationCircuit`? Or `BijectiveAggregationCircuit` if we want to feel fancy?

Reply:
TaikoAggregationCircuit seems good enough, changed.